Remove obsolete documentation (#342)
spacelift-io/user-documentation: docs/vendors/terraform/terraform-provider.md

Our Terraform provider is open source and its [README](https://github.com/spacelift-io/terraform-provider-spacelift) always contains the latest available documentation. It's also distributed as part of our [Docker runner image](../../integrations/docker.md#standard-runner-image) and available through our [own provider registry](terraform-provider.md#how-it-works). The purpose of this article isn't as much to document the provider itself but to show how it can be used to incorporate Spacelift resources into your infra-as-code.

> An example of a Spacelift stack being defined through Terraform:

Reviewer: Just a nit, this sentence comes across as a bit abrupt. The advantage of the "So, without further ado, ..." part was that it connected the example to the previous piece of text.

If you're going to reword it, I think it would need to be something like: "Here is an example showing how to define a Spacelift stack using Terraform".

}
```

> That's awesome. But can we put Terraform to good use and integrate it with resources from a completely different provider? Sure we can, and we have a good excuse, too. Stacks accessibility can be managed [by GitHub teams](../../concepts/stack/README.md#access-readers-and-writers-teams), so why don't we define some?

Reviewer 2: Don't we want to replace this with an up-to-date example instead of deleting it?

Without this, the "Taking it for a spin" section is underwhelming. 😄

Author: Can we point to the terraform starter guide on GitHub?
I think that's the only hands-on terraform guide that we have that guides you through using the Spacelift provider.

}
```

> A more in depth guide that will guide you through creating multiple types of Spacelift resources through Terraform can be found at [the terraform-starter repository](https://github.com/spacelift-io/terraform-starter).

Reviewer 2: I would suggest a slight edit, other than that it looks good to me.

```suggestion
A more in-depth guide to creating multiple types of Spacelift resources with Terraform can be found at [the terraform-starter repository](https://github.com/spacelift-io/terraform-starter){: rel="nofollow"}.
```
